### PR TITLE
feat: unsafeStatic hack

### PR DESCRIPTION
**Pull request by aleespryker**
Closes HRZ-2280

Still WIP; don't merge until HRZ-2225 is merged.

### CODE REVIEW

**Comment by AndreyYevtukhov** [comment]
@aleespryker conflicts with `dev`

**Comment by aleespryker** [comment]
> @aleespryker conflicts with `dev`

Fixed

File: libs/domain/auth/src/services/components.provider.ts [hunk 1]
```diff
@@ -1,24 +1,7 @@
...
-    },
-    ['auth-logout']: {
-      template: (uid: string, layoutClasses?: string) =>
-        html`<auth-logout
-          uid="${uid}"
-          class=${ifDefined(layoutClasses)}
-        ></auth-logout>`,
-    },
-  },
+  useValue: true,
```
Comment (dunqan) [inline libs/domain/auth/src/services/components.provider.ts]: we should just drop it

File: libs/domain/cart/src/services/components.provider.ts [hunk 2]
```diff
@@ -1,38 +1,7 @@
...
-    },
-    ['cart-summary']: {
-      template: (uid: string, layoutClasses?: string) =>
-        html`<oryx-cart-summary
-          uid="${uid}"
-          class=${ifDefined(layoutClasses)}
-        ></oryx-cart-summary>`,
-    },
-  },
+  useValue: true,
```
Comment (dunqan) [inline libs/domain/cart/src/services/components.provider.ts]: Let's remove all those providers

File: libs/platform/experience/src/services/registry/default-components-registry.service.ts [hunk 3]
```diff
@@ -32,7 +32,14 @@ export class DefaultComponentsRegistryService
     uid: string,
     styleClasses?: string
   ): TemplateResult | undefined {
-    const component = this.componentMapping?.[type];
+    let component;
```
Comment (dunqan) [inline libs/platform/experience/src/services/registry/default-components-registry.service.ts]: component mappig should have precedence, and only if there is no explicit mapping defined, we want to reach to component plugin, to check, if it provides this component.

Comment (dunqan) [reply libs/platform/experience/src/services/registry/default-components-registry.service.ts]: but actually, let's simplify:
- don't check component plugin at all (bring back protected method)
- just render the type as tag always, even if we don't have a mapping

**Review by dunqan — CHANGES_REQUESTED** [review]

File: libs/platform/core/src/orchestration/components/components-loader.ts [hunk 4]
```diff
@@ -42,7 +42,7 @@ export class ComponentsLoader {
     }
   }
 
-  protected findComponentDefBy(name: string): ComponentDef {
+  findComponentDefBy(name: string): ComponentDef {
```
Comment (dunqan) [inline libs/platform/core/src/orchestration/components/components-loader.ts]: can we revert this?

File: libs/template/presets/src/features/b2c/experience/header.ts [hunk 5]
```diff
@@ -65,9 +65,8 @@ export const HeaderTemplate: StaticComponent = {
           },
         },
         { type: 'oryx-user-summary' },
-        { type: 'auth-logout' },
```
Comment (dunqan) [inline libs/template/presets/src/features/b2c/experience/header.ts:68]: is it deliberate?

File: libs/platform/experience/src/services/registry/default-components-registry.service.ts [hunk 6]
```diff
@@ -13,13 +13,12 @@ export class DefaultComponentsRegistryService
   protected componentMapping?: ComponentMapping;
 
   constructor(
-    registeredComponents = inject(ComponentMapping),
+    registeredComponents = inject(ComponentMapping, []),
```
Comment (dunqan) [inline libs/platform/experience/src/services/registry/default-components-registry.service.ts]: let's use `[] as ...` to fix typings instead

**Review by dunqan — CHANGES_REQUESTED** [review]

File: libs/template/presets/src/features/b2c/experience/header.ts (hunk 5 above)
Comment (aleespryker) [reply libs/template/presets/src/features/b2c/experience/header.ts:68]: We are using `oryx-auth-logout` now. Also noticed that `oryx-user-summary` includes the logout component anyways, so auth-logout is not needed. I asked @tobi-or-not-tobi about it earlier.

**Review by dunqan — APPROVED** [review]

**Pull request merged**